nixos/maintainers/scripts/ec2/amazon-image.nix

```diff
@@ -51,7 +51,9 @@ in {
     inherit lib config;
     inherit (cfg) contents format name;
     pkgs = import ../../../.. { inherit (pkgs) system; }; # ensure we use the regular qemu-kvm package
-    partitionTableType = if config.ec2.hvm then "legacy" else "none";
+    partitionTableType = if config.ec2.efi then "efi"
+                         else if config.ec2.hvm then "legacy"
```

Comment:
`ec2.hvm` is blocked by an assertion, but the implementation still seems to support it. I preserved the existing behavior.